### DISCUSSION TITLE
Enforcing static readonly on <const from-guid> - Wrapper class

### DISCUSSION THREAD

**Discussion by h1cks** [SharpGenTools/SharpGenTools#213] [Q&A] [answered]
HI,

Will try to articulate this as well as i can, I'm at the moment trying to work through adding mappings for Vortice Windows.  My issue is that when using a generic wrapper class, the mapping if not a system.guid will resolve to a const rather than static readonly declaration.

To demonstrate my point, i'm leveraging alot of what sharpdx did around this in https://github.com/sharpdx/SharpDX/blob/master/Source/SharpDX.MediaFoundation/Mapping-core.xml
(line 137 is my example below).

using the mapping rule to map to the generic 

```
 <const from-guid="MF_MEDIA_ENGINE_AUDIO_CATEGORY" class="SharpDX.MediaFoundation.MediaEngineAttributeKeys" type="MediaAttributeKey&lt;SharpDX.Multimedia.AudioStreamCategory&gt;" name="AudioCategory">new MediaAttributeKey&lt;SharpDX.Multimedia.AudioStreamCategory&gt;("$1", "AudioCategory")</const>
```

The code created is 

```
public const MediaAttributeKey<Vortice.Multimedia.AudioStreamCategory> AudioCategory = unchecked((MediaAttributeKey<Vortice.Multimedia.AudioStreamCategory>)(new MediaAttributeKey<Vortice.Multimedia.AudioStreamCategory>("c8d4c51d-350e-41f2-ba46-faebbb0857f6", "AudioCategory")));
```

This complains obviously that you can not assign a run time creation at compile time which is obvious after generation.  So, just wondering if there is a way to force static readonly instead?   As this sorts the issue, if not, no problem either.  

Apologies, new to this, so my question may be elementary or completely awkward.

My work around is to copy the code out ,create a new class and remove the mappings.  Which i'm fine with if at the end of the day the code creation has got me 90% of the way.

Thanks

**Comment by andrew-boyarshin** [comment, marked as answer]
`visibility="public static readonly"` attribute should help.

**Comment by h1cks** [comment]
cheers, will try it out.
